src/ServerPrivate.cc

```diff
@@ -436,6 +453,49 @@ bool ServerPrivate::WorldsService(ignition::msgs::StringMsg_V &_res)
   return true;
 }
 
+//////////////////////////////////////////////////
+bool ServerPrivate::ServerControlService(
+  const ignition::msgs::ServerControl &_req, msgs::Boolean &_res)
+{
+  _res.set_data(false);
+
+  if (_req.stop())
+  {
+    if (!this->stopThread)
+    {
+      this->stopThread.reset(new std::thread([this]{
+        ignlog << "Stopping Gazebo" << std::endl;
+        std::this_thread::sleep_for(std::chrono::milliseconds(1));
+        this->Stop();
+      }));
+    }
+    _res.set_data(true);
+  }
+
+  // TODO(chapulina): implement world cloning
+  if (_req.clone() || _req.new_port() != 0 || !_req.save_world_name().empty())
+  {
+    ignerr << "ServerControl::clone is not implemented" << std::endl;
+    _res.set_data(false);
```

Comment:
This is already set up-top, so I think we don't need to repeat it in every block

Reply:
The individual parts are not `else if`s, but `if`s, so theoretically, the user could ask for more functions in one call. The logic I used was to report the worst result of all, i.e. report `false` as soon as any of the requested functionalities fails. It is set to false at the beginning, but I added these to "remind" or make sure that a failing function is reported. However, it might be wiser to turn the `if`s to `else if`s, as nobody probably wants to simultaneously stop the server and clone/create a world.

Reply:
Ah gotcha, I think future-proofing is ok